utils/make_release_notes.sh

```diff
@@ -0,0 +1,120 @@
+
+NOTE: all version strings must already be existing tags in both the Cyclus, Cycamore, and Cymetric repositories.
+
+The following environment variables must be set for this script to function:
+
+     CYCLUS_DIR : Environment variable CYCLUS_DIR must be set to the cyclus repository directory.
+   CYCAMORE_DIR : Environment variable CYCAMORE_DIR must be set to the cycamore repository directory.
+   CYMETRIC_DIR : Environment variable CYMETRIC_DIR must be set to the cymetric repository directory.
+
+EOF
+
+}
+
+die() {
+    echo -n >&2 "ERROR: "
+    echo >&2 "$@"
+    usage
+    exit 1
+}
+
+# special behavior for macosx
+sed_i() {
+    if [[ "`uname`" == "Linux" ]]; then
+	sed -i  "$1" "$2";
+    else
+    	sed -i '' "$1" "$2";
+    fi
+}
+
+HERE=$PWD
+
+# check input
+CYCLUS=${CYCLUS_DIR?"Environment variable CYCLUS_DIR must be set to the cyclus repository directory."}
+CYCAMORE=${CYCAMORE_DIR?"Environment variable CYCAMORE_DIR must be set to the cycamore repository directory."}
+CYMETRIC=${CYMETRIC_DIR?"Environment variable CYMETRIC_DIR must be set to the cymetric repository directory."}
+[ "$#" -eq 2 ] || die "Must provide the from version and to version (e.g., W.W.W -> X.X.X) as an argument"
+
+PREV=$1
+VERSION=$2
+echo "Making release notes template for Cyclus stack version $VERSION from $PREV. 
+
+Note, these version names must refer to *existing tags* in all repositories.
+"
+
+# cyclus summary
+cd $CYCLUS
+NCOMMITS=`git rev-list $PREV...$VERSION --count | tail -n1`
+SUMMARY=`git diff --stat $PREV...$VERSION | tail -n1`
+CYCLUSTXT="$NCOMMITS commits resulting in $SUMMARY"
+CYCLUSCONTRIB=`git log --format='%aN' $PREV...$VERSION | sort -u`
+cd $HERE
+
+# cycamore summary
+cd $CYCAMORE
+NCOMMITS=`git rev-list $PREV...$VERSION --count | tail -n1`
+SUMMARY=`git diff --stat $PREV...$VERSION | tail -n1`
+CYCAMORETXT="$NCOMMITS commits resulting in $SUMMARY"
+CYCAMORECONTRIB=`git log --format="%aN" $PREV...$VERSION | sort -u`
+cd $HERE
+
+# cymetric summary
+cd $CYMETRIC
+[[ `git tag` =~ $PREV ]] && CYMETRIC_PREV=$PREV || CYMETRIC_PREV=`git log --pretty=format:%H|tail -1`
+NCOMMITS=`git rev-list $CYMETRIC_PREV...$VERSION --count | tail -n1`
+SUMMARY=`git diff --stat $CYMETRIC_PREV...$VERSION | tail -n1`
+CYMETRICTXT="$NCOMMITS commits resulting in $SUMMARY"
+CYMETRICCONTRIB=`git log --format='%aN' $CYMETRIC_PREV...$VERSION | sort -u`
+cd $HERE
+
+# contributors, beware, thar be hackery ahead
+echo "Raw Cyclus core contributors:"
+echo "$CYCLUSCONTRIB"
+echo ""
+echo "$CYCLUSCONTRIB" > .contribs
+echo "Raw Cycamore contributors:"
+echo "$CYCAMORECONTRIB"
+echo ""
+echo "$CYCAMORECONTRIB" >> .contribs
+echo "Raw Cymetric core contributors:"
+echo "$CYMETRICCONTRIB"
+echo ""
+echo "$CYMETRICCONTRIB" > .contribs
+CONTRIBTXT=`cat .contribs | sort -u | awk '{print "* " $0}'`
+echo "$CONTRIBTXT" > .contribs
+
+# replace
+FILE=release_notes.rst
+cp -i release_notes.rst.in $FILE
```

Comment:
I disagree. This is a script that requires updating after it is run. These
are only helper scripts during the release process.

On Fri, May 8, 2015 at 7:06 PM, Anthony Scopatz <notifications@github.com>
wrote:

> In utils/make_release_notes.sh
> <https://github.com/cyclus/ciclus/pull/63#discussion_r29986239>:
>
> > +echo ""
> > +echo "$CYCLUSCONTRIB" > .contribs
> > +echo "Raw Cycamore contributors:"
> > +echo "$CYCAMORECONTRIB"
> > +echo ""
> > +echo "$CYCAMORECONTRIB" >> .contribs
> > +echo "Raw Cymetric core contributors:"
> > +echo "$CYMETRICCONTRIB"
> > +echo ""
> > +echo "$CYMETRICCONTRIB" > .contribs
> > +CONTRIBTXT=`cat .contribs | sort -u | awk '{print "* " $0}'`
> > +echo "$CONTRIBTXT" > .contribs
> > +
> > +# replace
> > +FILE=release_notes.rst
> > +cp -i release_notes.rst.in $FILE
>
> Then this should be optional. Scripts should always be runnable in a fully
> automated way. Maybe add an optional -i flag to the script itself.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/ciclus/pull/63/files#r29986239>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192